packages/x-components/src/x-modules/history-queries/components/toggle-history-queries.vue

```diff
@@ -0,0 +1,78 @@
+<template>
+  <BaseSwitch @change="toggle" :value="isEnabled" />
+</template>
+
+<script lang="ts">
+  import Vue from 'vue';
+  import { Component } from 'vue-property-decorator';
+  import BaseSwitch from '../../../components/base-switch.vue';
+  import { State, xComponentMixin } from '../../../components';
+  import { historyQueriesXModule } from '../x-module';
+
+  /**
+   * Toggle history queries component to enable or disable the history queries.
+   * This component emits events depending on the `isEnabled` value.
+   *
+   * @public
+   */
+  @Component({
+    mixins: [xComponentMixin(historyQueriesXModule)],
+    components: { BaseSwitch }
+  })
+  export default class ToggleHistoryQueries extends Vue {
```

Comment:
Should we name it `HistoryQueriesSwitch`? Just to match the control that is being rendered

Reply:
Sure. I'll rename it.